### ISSUE TITLE
piping weight_draws

### ISSUE THREAD

**Issue by avehtari** [stan-dev/posterior#310]
Currently `weight_draws(x, weights, ...)` requires `weights` to be a numeric vector. Would it be possible to have a piping version so that when the draws object already has a variable that contains the log weights e.g. named `lw`, we could pipe
```
x |> mutate_varibles(lw = lp__-lp_approx__) |> weight_draws(x, weights=lw, log=TRUE)
```

and even better if we can pipe as
```
x |> weight_draws(x, weights=lp__-lp_approx__, log=TRUE)
```

**Comment by paul-buerkner** [comment]
It all depends on where we search for the weights vector. Currently, we search for it in the parent environment but we could of course change this to search in the draws object first and only then in the parent environment. This is not really related to whether we pipe or not, but in any case making this work would be a backwards compatibility breaking change that we should approach carefully.

**Comment by avehtari** [comment]
I think this is related to piping in that way that the idea of piping is that we don't need to store temporary variables in the parent environment. I understand the possibility of breaking something, but I assume not many have used these functions yet.

Now that CmdStanR supports Laplace and Pathfinder, I'm going to use `weight_draws()` and `resample_draws()` often, but then would like to have a simple looking approach.

**Comment by avehtari** [comment]
I guess this is what I need to do now (assuming draws in matrix format)
```
draws8rhs <- draws8rhs |> mutate_variables(lw=lp__-lp_approx__)
draws8rhs1 <- draws8rhs|> 
    weight_draws(weights=extract_variable(draws8rhs,"lw"), log=TRUE)|>
    resample_draws(ndraws=1)
```

**Comment by paul-buerkner** [comment]
Yes, good point. We should definitely consider the weights argument to search for variables in the draws object first.

As for the current approach, the following short work too, I think (not tested myself):

```R
draws8rhs <- draws8rhs |> 
  mutate_variables(lw=lp__-lp_approx__) |> 
  weight_draws(weights=extract_variable(draws8rhs,"lw"), log=TRUE) |>
  resample_draws(ndraws=1)
```

**Comment by avehtari** [comment]
That doesn't work 
```
+ Error in `[.default`(x, , i2, drop = FALSE) : subscript out of bounds
```

**Comment by paul-buerkner** [comment]
Ah right. Makes sense. Overlooked that one. Your approach is probably the currently sensible one and I agree it's ugly.
